addons/mail/static/src/models/message_pending/message_pending.js

```diff
@@ -0,0 +1,203 @@
+                const target = `target='_blank'`;
+                const link = `<a ${href} ${attClass} ${dataOeId} ${dataOeModel} ${target}>${mention.text}</a>`;
+                body = body.replace(mention.placeholder, link);
+            }
+            return body;
+        }
+
+        /**
+         * @private
+         * @param {string} htmlString
+         * @returns {string}
+         */
+        _generateEmojisOnHtml(htmlString) {
+            for (const emoji of emojis) {
+                for (const source of emoji.sources) {
+                    const escapedSource = String(source).replace(
+                        /([.*+?=^!:${}()|[\]/\\])/g,
+                        '\\$1');
+                    const regexp = new RegExp(
+                        '(\\s|^)(' + escapedSource + ')(?=\\s|$)',
+                        'g');
+                    htmlString = htmlString.replace(regexp, '$1' + emoji.unicode);
+                }
+            }
+            return htmlString;
+        }
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsLog() {
+            return this.composer.isLog;
+        }
+
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeThread() {
+            return replace(this.composer.thread);
+        }
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeRawBody() {
+            return this.composer.textInputContent;
+        }
+
+    }
+
+    PendingMessage.fields = {
+        /**
+         * States which attachments are currently being created in this pending message.
+         */
+        attachments: many2many('mail.attachment', {
+            inverse: 'pendingMessages',
+        }),
+        /**
+         * Determines the body that will be sent to the server.
+         * It will be created by the `generateBody` function when needed to
+         * avoid performance issue
+         */
+        body: attr(),
+        /**
+         * States the body without any transformation. It basicly the content of
+         * the composer.
+         */
+        rawBody: attr({
+            compute: '_computeRawBody',
+        }),
+        /**
+         * States the mentionned channel in the rawBody.
+         */
+        mentionedChannels: many2many('mail.thread'),
+        /**
+         * States the mentionned partners in the rawBody.
+         */
+        mentionedPartners: many2many('mail.partner'),
+        thread: one2one('mail.thread', {
+            compute: '_computeThread',
+        }),
+        isLog: attr({
+            compute: '_computeIsLog',
+        }),
+        recipients: many2many('mail.partner', {
+            compute: '_computeRecipients',
+        }),
+        composer: one2one('mail.composer', {
+            inverse: 'pendingMessage',
+            readonly: true,
+            required: true,
+        }),
+    };
+    PendingMessage.identifyingFields = ['composer'];
```

Comment:
I'm not fond of name "PendingMessage" for this model: it's not a message (and no relation to message), and "pending" is too broad.

I suggest "MessageComposition".